synapse/storage/util/id_generators.py

```diff
@@ -91,7 +91,14 @@ class StreamIdGenerator:
             # ... persist event ...
     """
 
-    def __init__(self, db_conn, table, column, extra_tables=[], step=1):
+    def __init__(
+        self,
+        db_conn,
+        table,
+        column,
+        extra_tables: Iterable[Tuple] = frozenset(),
```

Comment:
why not use a zero-length tuple here?

```suggestion
        extra_tables: Iterable[Tuple[str, str]] = (),
```